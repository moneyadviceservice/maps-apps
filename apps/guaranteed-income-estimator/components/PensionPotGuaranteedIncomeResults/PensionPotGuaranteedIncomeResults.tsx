import { NumericFormat } from 'react-number-format';

import { H2 } from '@maps-react/common/components/Heading';
import { useTranslation } from '@maps-react/hooks/useTranslation';
import { PensionPotCalculatorResults } from '@maps-react/pension-tools/types';
import { formatQuery } from '@maps-react/pension-tools/utils/formatQuery';

import { SharedResultsHeading } from '../../data/form-content/text/guaranteedIncomeEstimator';
import { guaranteedIncomeCalculator } from '../../utils/guaranteedIncomeCalculator';

export const PensionPotGuaranteedIncomeResults = ({
  queryData,
  data,
}: Omit<PensionPotCalculatorResults, 'fields, onChange'>) => {
  const { z } = useTranslation();

  const results = guaranteedIncomeCalculator(
    formatQuery(queryData.pot),
    formatQuery(queryData.age),
  );

  return (
    <div id="results">
      <H2 className="text-blue-700 mb-6 md:mb-8">{data.resultTitle}</H2>
      {results && (
        <dl>
          <SharedResultsHeading
            pot={queryData.pot}
            taxFreeLumpSum={results.taxFreeLumpSum}
            text={z({
              en: 'pot in one go, you could get:',
              cy: 'i brynu blwydd-dal, gallech gael:',
            })}
          />
          <dt className="mb-2 font-medium">
            {z({
              en: 'and get:',
              cy: 'a chael:',
            })}
          </dt>
          <dd className="mb-4 text-4xl font-bold">
            <NumericFormat
              value={results.income}
              prefix="Â£"
              thousandSeparator=","
              displayType="text"
            />{' '}
            {z({
              en: 'guaranteed annual taxable income, for the rest of your life',
              cy: 'o incwm trethadwy blynyddol gwarantedig, am weddill eich oes',
            })}
          </dd>
        </dl>
      )}
      <div className="mt-8 text-base">{data.calloutMessageResults}</div>
    </div>
  );
};
