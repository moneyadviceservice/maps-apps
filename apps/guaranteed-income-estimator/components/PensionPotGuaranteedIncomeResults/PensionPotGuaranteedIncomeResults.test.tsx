import { render } from '@testing-library/react';

import { PensionPotGuaranteedIncomeResults } from './PensionPotGuaranteedIncomeResults';
jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    push: jest.fn(),
    query: {
      language: 'en',
    },
    events: {
      on: jest.fn(),
      off: jest.fn(),
    },
  }),
}));

const mockQueryData = {
  pot: '100000',
  age: '65',
};
export const mockData = {
  title: 'Guaranteed income Calculator',
  errorTitle: "There's been a problem",
  buttonText: 'Calculate',
  submittedButtonText: 'Recalculate',
  resultsButtonText: 'Apply changes',
  resultTitle: 'Your results',
};

describe('PensionPotGuaranteedIncomeResults', () => {
  it('should render correctly', () => {
    const { container } = render(
      <PensionPotGuaranteedIncomeResults
        queryData={mockQueryData}
        data={mockData}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
