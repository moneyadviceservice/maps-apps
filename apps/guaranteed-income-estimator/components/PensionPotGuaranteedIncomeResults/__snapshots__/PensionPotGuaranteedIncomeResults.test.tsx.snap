// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PensionPotGuaranteedIncomeResults should render correctly 1`] = `
<div>
  <div
    id="results"
  >
    <h2
      class="text-3xl md:text-5xl font-bold text-blue-700 mb-6 md:mb-8"
    >
      Your results
    </h2>
    <dl>
      <dt
        class="mb-2 font-medium"
      >
        Based on what you’ve told us, if you use your
         
        <span>
          £100,000
        </span>
         
        pot in one go, you could get:
      </dt>
      <dd
        class="mb-4 text-4xl font-bold"
      >
        <span>
          £25,000
        </span>
         
        tax free
      </dd>
      <dt
        class="mb-2 font-medium"
      >
        and get:
      </dt>
      <dd
        class="mb-4 text-4xl font-bold"
      >
        <span>
          £5,600
        </span>
         
        guaranteed annual taxable income, for the rest of your life
      </dd>
    </dl>
    <div
      class="mt-8 text-base"
    />
  </div>
</div>
`;
